differ/mutationDiffer.go

```diff
@@ -756,79 +764,89 @@ func (dw *DifferWorker) diff() {
+				if bodyOnly {
+					srcerr = sourceResult.bodyErr
+					tgterr = targetResult.bodyErr
+				} else {
+					srcerr = sourceResult.metaErr
+					tgterr = targetResult.metaErr
+				}
+				if isKeyNotFoundError(srcerr) && !isKeyNotFoundError(tgterr) {
 					if _, exists := missingFromSource[srcColId]; !exists {
 						missingFromSource[srcColId] = make(map[string]*GocbResult)
 					}
-					missingFromSource[srcColId][key] = gocbResultConstructor(targetResult.GoCbResult())
+					missingFromSource[srcColId][key] = gocbResultConstructor(targetResult)
 					continue
 				}
-				if !isKeyNotFoundError(sourceResult.Error()) && isKeyNotFoundError(targetResult.Error()) {
+				if !isKeyNotFoundError(srcerr) && isKeyNotFoundError(tgterr) {
 					if _, exists := missingFromTarget[tgtColId]; !exists {
 						missingFromTarget[tgtColId] = make(map[string]*GocbResult)
 					}
-					missingFromTarget[tgtColId][key] = gocbResultConstructor(sourceResult.GoCbResult())
+					missingFromTarget[tgtColId][key] = gocbResultConstructor(sourceResult)
 					continue
 				}
-				if !areResultsTheSame(sourceResult.GoCbResult(), targetResult.GoCbResult()) {
-					if isDeletedPerMetadata != nil && isDeletedPerMetadata(sourceResult.GoCbResult()) {
-						if _, exists := deletedFromSource[srcColId]; !exists {
-							deletedFromSource[srcColId] = make(map[string][]*GocbResult)
+				if bodyOnly {
+					if !areGetResultsBodyTheSame(sourceResult, targetResult) {
+						if _, exists := srcDiff[srcColId]; !exists {
+							srcDiff[srcColId] = make(map[string][]*GocbResult)
 						}
-						deletedFromSource[srcColId][key] = append(deletedFromSource[srcColId][key], []*GocbResult{gocbResultConstructor(sourceResult.GoCbResult()), gocbResultConstructor(targetResult.GoCbResult())}...)
-						continue
-					}
-					if isDeletedPerMetadata != nil && isDeletedPerMetadata(targetResult.GoCbResult()) {
-						if _, exists := deletedFromTarget[tgtColId]; !exists {
-							deletedFromTarget[tgtColId] = make(map[string][]*GocbResult)
+						srcDiff[srcColId][key] = append(srcDiff[srcColId][key], []*GocbResult{gocbResultConstructor(sourceResult), gocbResultConstructor(targetResult)}...)
+						if _, exists := tgtDiff[tgtColId]; !exists {
+							tgtDiff[tgtColId] = make(map[string][]*GocbResult)
 						}
-						deletedFromTarget[tgtColId][key] = append(deletedFromTarget[tgtColId][key], []*GocbResult{gocbResultConstructor(sourceResult.GoCbResult()), gocbResultConstructor(targetResult.GoCbResult())}...)
-						continue
+						tgtDiff[tgtColId][key] = append(tgtDiff[tgtColId][key], []*GocbResult{gocbResultConstructor(targetResult), gocbResultConstructor(sourceResult)}...)
 					}
-					if _, exists := srcDiff[srcColId]; !exists {
-						srcDiff[srcColId] = make(map[string][]*GocbResult)
+				} else {
+					srcUUID, err := hlv.UUIDtoDocumentSource(dw.differ.sourceBucketUUID)
+					if err != nil {
+						dw.logger.Errorf("Error in converting the bucket UUID to required HLV format. err: %v", err)
```

Comment:
Log the document key

Reply:
This error is independent of the document key being compared.
So I have moved this code block outside the for loop(used for iterating over the fetch list) for better readability